### ISSUE TITLE
I'm pretty sure that the authors used test datasets while training.

### ISSUE THREAD

**Issue by CauchyComplete** [Knightzjz/NCL-IML#12]
I am leaving this issue for future researchers, in case it is helpful.

There appears to be a significant issue with the experimental setup described in this paper.

In the academic community, it has become standard practice—at least since 2021—to evaluate IFL experiments using entirely non-overlapping training and test sets. However, according to Section 4 (Datasets), this paper appears to have merged several datasets and then split them into train and test sets.

As a result, **the comparisons in Table 1 are meaningless**, since other methods (may not be all, but at least MVSS-Net) ensure there is no overlap between train and test data.

The paper repeatedly points out that other methods rely on large-scale pretraining data, whereas the authors did not. However, in this case, it seems that the test set has been partially used for training.

This may have led to an overestimation of performance compared to other recent methods, which strictly separate train and test sets. Based on their later work (such as the IMDL-BenCo benchmark), it is clear that the authors are well aware of the appropriate train/test protocols.

When this paper was first published, there were some uncertainties about these points, but I waited for further updates from the authors.

In [Issue #9](https://github.com/Knightzjz/NCL-IML/issues/9), the authors stated in December 2023 that the released weights were trained only on NC16, and that test results would therefore differ from those reported in the paper. However, as of now, properly trained weights have not yet been shared.

Additionally, while other methods developed by the authors have been implemented in their benchmark, the training code for NCL has still not been released. This further raises questions about the transparency of the methodology.

I believe it would be beneficial for the research community if the authors could clarify the train/test protocol used in the experiments and release the corresponding training code.
